### PR TITLE
fix: env port

### PR DESCRIPTION
**Pull request by miamai**
- run port in `8000`

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import mongoose from 'mongoose';
 import authRouter from './route/auth.route';
 
 const app = express();
-const port = process.env.PORT ?? 3000;
+const port = process.env.PORT || 8000;
```
Comment (fufuShih) [inline src/app.ts]:
```suggestion
const port = process.env.PORT ?? 8000;
```

Recommend to use `??` when dealing with env variables.

For example, if port is on 0:
```
console.log(0 ?? 3000); // 0
console.log(0 || 3000); // 3000
```

See:
1. [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#description)
2. [Stack overflow discussion](https://stackoverflow.com/questions/61480993/when-should-i-use-nullish-coalescing-vs-logical-or)

Comment (cedarwud) [reply src/app.ts]: if .env file write PORT=
will throw new error when use ??

Comment (fufuShih) [reply src/app.ts]: In the case of `undefined` and `null`, they behave the same way. Please see reference
```
console.log(undefined || "not found") // "not found"
console.log(null      || "not found") // "not found"

console.log(undefined ?? "not found") // "not found"
console.log(null      ?? "not found") // "not found"
```

image

Comment (cedarwud) [reply src/app.ts]: because in .env you write PORT=
system will get an empty string ''
rather than null or undefined
image

Comment (fufuShih) [reply src/app.ts]: Using nullish is a stricter approach that allows you to quickly identify imply problems. 

For your example, if process.env.PORT defaults to an empty string (""), and you deploy to the cloud, you would immediately know that process.env.PORT has not been set. 

But if you use the logical OR operator (||), it won't throw any error and will use the default value, leading to hidden problem, which wrong port setting on cloud.

**Comment by cedarwud** [comment]
我目前只有在 heroku 跟 render 佈署的經驗，這二個環境的 PORT 都是系統去設定的，不太確定其他的環境如何?
所以這邊的寫法比較是針對本機開發環境，因為是不好的環境變數設定造成的問題，所以也沒有特別要堅持哪種寫法，可以看看 Mia 這邊有什麼想法我都 ok

File: src/app.ts (hunk 1 above)
Comment (miamai) [reply src/app.ts]: remove as discussed

**Review by fufuShih — APPROVED** [review]

**Review by cedarwud — APPROVED** [review]

**Pull request merged**